source/develop/release-management/features/network/autodefine-external-network.md

```diff
@@ -0,0 +1,101 @@
+Engine automatically creates an external network based on physical VM network
+with physical connection to it. 
+
+Related patches can be found on [gerrit topic:autodefine](https://gerrit.ovirt.org/#/q/topic:autodefine).
+
+### Owner
+
+- Feature Owner: Ales Musil (amusil)
+- E-mail: <amusil@redhat.com>
+
+### Benefit to oVirt
+
+Autodefine helps reducing administrative tasks needed with OVS switch type in 
+the cluster.
+
+## Prerequisites
+
+Cluster has to be configured with switch type OVS and valid external provider.
+The user needs to make sure that connection to external provider is valid.
+
+### Limitations
+
+* With OVS switch type in cluster, only external networks are allowed to 
+  be attached to the VM interface. 
+
+* Physical network linked to any external network can't be removed. 
+
+* Existing Clusters switch cannot be changed to OVS. Currently there is no way 
+  in the Engine to follow this change across all networks and VM interfaces.
+
+## Detailed description
+
+### Creating of new physical network
+
+When creating new network choose its name. Check the `VM network`. If vlan 
+tagging is needed, check the `Enable VLAN tagging` and put the tag into the 
+field. Please note that auto-define can be triggered only on top of network that
+is not marked as external. 
+
+![Creation of physical network](../../../../images/wiki/auto-define_create.png  "New physical network")
+
+After confirmation if all conditions were met, the external network connected to 
+the physical one, will be created simultaneously. The auto-defined external 
+network is named `external_name_of_physical_network` (The physical network name
+might be shortened so that auto-defined network name is not bigger than 256 
+characters). 
+
+![Auto-define network created](../../../../images/wiki/auto-define_created.png  "Auto-define external network")
+
+In case there are clusters with different default external providers the Engine 
+will create multiple external network for every provider and add number behind 
+this auto-defined network name. 
+
+## Implementation
+
+### Engine
+
+* Added new query `GetAllNetworksForVmByClusterId`. The query returns 
+  only external networks for VM, that requests this query under Cluster with 
+  OVS switch type. 
+
+* Validations for all limited operations with auto-defined external network and 
+  its linked physical network. According to [Autodefine External Network - Limitations](#limitations)
+
+* Added new command `AutodefineExternalNetwork`. This command is 
+  triggered every time new network is created. The command validates the 
+  incoming network and its parameters. Then it finds all clusters with OVS 
+  switch type attached to incoming network and clusters external providers.
+  The command continues with creating new external network and connecting it to 
+  the incoming physical network and attaching to clusters. The command might 
+  create multiple external networks (One for each external provider). 
```

Comment:
... networks, one for each default external provider in the data center.

Reply:
Done